lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1626,6 +1630,13 @@ struct FIRRTLLowering : public FIRRTLVisitor<FIRRTLLowering, LogicalResult> {
   /// A namespace that can be used to generte new symbol names that are unique
   /// within this module.
   hw::ModuleNamespace moduleNamespace;
+
+  /// A backedge builder to directly materialize values during the lowering
+  /// without requiring temporary wires.
+  BackedgeBuilder backedgeBuilder;
+  /// Currently unresolved backedges. More precisely, a mapping from the
+  /// backedge's value to the backedge itself.
+  llvm::SmallDenseMap<Value, Backedge> backedges;
```

Comment:
This seems generally useful. Maybe put it in backedgebuilder?

Reply:
Yeah that sounds like an idea! I'll take a look.